### PR TITLE
feat: Implement organization role assignments for both users and teams

### LINKED ISSUE #3280: feat: Support assigning organization roles to teams and users
The GitHub API supports assigning (and removing) an Organization role to a user/team. These are the endpoints that should be supported (including both adding and removing an assignment):
* [`/orgs/ORG/organization-roles/teams/TEAM_SLUG/ROLE_ID`](https://docs.github.com/en/enterprise-cloud@latest/rest/orgs/organization-roles?apiVersion=2022-11-28#assign-an-organization-role-to-a-team)
* [`/orgs/ORG/organization-roles/users/USERNAME/ROLE_ID`](https://docs.github.com/en/enterprise-cloud@latest/rest/orgs/organization-roles?apiVersion=2022-11-28#assign-an-organization-role-to-a-team)

I can work on this as well

### PR DESCRIPTION
**Pull request by felixlut**
resolves #3280

### CODE REVIEW

**Comment by felixlut** [comment]
While working on this I noticed that the file naming is slightly off from how GitHub structures their API:s. There's a difference between [repository roles](https://docs.github.com/en/enterprise-cloud@latest/rest/orgs/custom-roles?apiVersion=2022-11-28) and [organization roles](https://docs.github.com/en/enterprise-cloud@latest/rest/orgs/organization-roles?apiVersion=2022-11-28), but currently the `orgs_custom_roles.go` file is dealing with both (even before my changes). It should probably be split up into ``orgs_repo_roles.go`` and ``orgs_org_roles.go`` (or maybe flip the order to `orgs_role_{org/repo}.go` to keep the files "closer together"), similarily to https://github.com/integrations/terraform-provider-github/issues/2224. 

Leaving it up to the reviewer to decide if this should be done in this PR directly or in a separate later

**Comment by gmlewis** [comment]
> Leaving it up to the reviewer to decide if this should be done in this PR directly or in a separate one later

Please perform the reorganization in your next PR.  Thank you!

File: github/orgs_custom_roles.go [hunk 1]
```diff
@@ -156,6 +156,94 @@ func (s *OrganizationsService) DeleteCustomOrgRole(ctx context.Context, org stri
 	return resp, nil
 }
 
+// AssignOrgRoleToTeam assigns an existing organization role to a team in this organization.
+// In order to assign organization roles in an organization, the authenticated user must be an organization owner.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/organization-roles#assign-an-organization-role-to-a-team
+//
+//meta:operation PUT /orgs/{org}/organization-roles/teams/{team_slug}/{role_id}
+func (s *OrganizationsService) AssignOrgRoleToTeam(ctx context.Context, org string, teamSlug string, roleID int64) (*Response, error) {
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:165]:
```suggestion
func (s *OrganizationsService) AssignOrgRoleToTeam(ctx context.Context, org, teamSlug string, roleID int64) (*Response, error) {
```

File: github/orgs_custom_roles.go [hunk 2]
```diff
@@ -156,6 +156,94 @@ func (s *OrganizationsService) DeleteCustomOrgRole(ctx context.Context, org stri
...
+	return resp, nil
+}
+
+// RemoveOrgRoleFromTeam removes an existing organization role assignment to a team in this organization.
+// In order to remove organization role assignments in an organization, the authenticated user must be an organization owner.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/organization-roles#remove-an-organization-role-from-a-team
+//
+//meta:operation DELETE /orgs/{org}/organization-roles/teams/{team_slug}/{role_id}
+func (s *OrganizationsService) RemoveOrgRoleFromTeam(ctx context.Context, org string, teamSlug string, roleID int64) (*Response, error) {
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:187]:
```suggestion
func (s *OrganizationsService) RemoveOrgRoleFromTeam(ctx context.Context, org, teamSlug string, roleID int64) (*Response, error) {
```

File: github/orgs_custom_roles.go [hunk 3]
```diff
@@ -156,6 +156,94 @@ func (s *OrganizationsService) DeleteCustomOrgRole(ctx context.Context, org stri
...
+	return resp, nil
+}
+
+// AssignOrgRoleToUser assigns an existing organization role to a user in this organization.
+// In order to assign organization roles in an organization, the authenticated user must be an organization owner.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/organization-roles#assign-an-organization-role-to-a-user
+//
+//meta:operation PUT /orgs/{org}/organization-roles/users/{username}/{role_id}
+func (s *OrganizationsService) AssignOrgRoleToUser(ctx context.Context, org string, username string, roleID int64) (*Response, error) {
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:209]:
```suggestion
func (s *OrganizationsService) AssignOrgRoleToUser(ctx context.Context, org, username string, roleID int64) (*Response, error) {
```

File: github/orgs_custom_roles.go [hunk 4]
```diff
@@ -156,6 +156,94 @@ func (s *OrganizationsService) DeleteCustomOrgRole(ctx context.Context, org stri
...
+
+	resp, err := s.client.Do(ctx, req, nil)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, nil
+}
+
+// RemoveOrgRoleFromUser removes an existing organization role assignment to a user in this organization.
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:225]:
```suggestion
// RemoveOrgRoleFromUser removes an existing organization role assignment from a user in this organization.
```

File: github/orgs_custom_roles.go [hunk 5]
```diff
@@ -156,6 +156,94 @@ func (s *OrganizationsService) DeleteCustomOrgRole(ctx context.Context, org stri
...
+
+	resp, err := s.client.Do(ctx, req, nil)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, nil
+}
+
+// RemoveOrgRoleFromTeam removes an existing organization role assignment to a team in this organization.
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:181]:
```suggestion
// RemoveOrgRoleFromTeam removes an existing organization role assignment from a team in this organization.
```

File: github/orgs_custom_roles.go [hunk 6]
```diff
@@ -156,6 +156,94 @@ func (s *OrganizationsService) DeleteCustomOrgRole(ctx context.Context, org stri
...
+	return resp, nil
+}
+
+// RemoveOrgRoleFromUser removes an existing organization role assignment to a user in this organization.
+// In order to remove organization role assignments in an organization, the authenticated user must be an organization owner.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/organization-roles#remove-an-organization-role-from-a-user
+//
+//meta:operation DELETE /orgs/{org}/organization-roles/users/{username}/{role_id}
+func (s *OrganizationsService) RemoveOrgRoleFromUser(ctx context.Context, org string, username string, roleID int64) (*Response, error) {
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:231]:
```suggestion
func (s *OrganizationsService) RemoveOrgRoleFromUser(ctx context.Context, org, username string, roleID int64) (*Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @felixlut !
Just a few minor tweaks, please, then we should be ready for a second LGTM+Approval from any other contributor to this repo.

**Comment by felixlut** [comment]
@gmlewis done!

**Review by gmlewis — APPROVED** [review]
Thank you, @felixlut !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by tomfeigin — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @tomfeigin !
Merging.

**Pull request merged**